csharp/rust/src/data.rs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright Valkey GLIDE Project Contributors - SPDX Identifier: Apache-2.0
+
+use crate::value::Value;
+use std::ffi::{
+    c_int, c_void,
+};
+use std::fmt::{Display, Formatter};
+use std::os::raw::c_char;
+use std::ptr::null;
+use std::str::Utf8Error;
+use tokio::runtime::Runtime;
+
+pub struct FFIHandle {
+    pub runtime: Runtime,
+    pub handle: crate::apihandle::Handle,
+}
+
+#[repr(C)]
+#[derive(PartialEq)]
+pub struct CreateClientHandleResult {
+    pub result: ECreateClientHandleCode,
+    pub client_handle: *const c_void,
+    pub error_string: *const c_char,
+}
+
+#[repr(C)]
+#[derive(PartialEq, Eq, Debug, Clone, Copy, Hash)]
+pub enum ECreateClientHandleCode {
+    Success = 0,
```

Comment:
Success or ConnectionError
I don't like such limiting, but we have to follow that

Reply:
This will cause problems when actually using valkey glide. I am highly against limiting the possibility to handle the exceptions in the dotnet code properly, knowing what actually happened (timeout and connection refused errors are majorly different errors). Removing this will leave the enduser in the dark about what actually happened.